docs/source/torch.compiler_deepdive.rst

```diff
@@ -210,53 +210,128 @@ point for the first time.
-       # we use python interpreter to select the branch
-       # depending on the value, the rest graph is either `__resume_at_30_1`
-       # or `__resume_at_38_2`
-       if b_sum_less_than_0:
+   from depyf import decompile
+   print("guard code:")
+   print(decompile(guard))
+   print("compiled code:")
+   print(decompile(code))
+
+The output is:
+
+::
+
+   guard code:
+   def guard(L):
+       if not getattr(___guarded_code, 'valid'):
+           return False
+       _var0 = L['a']
+       if not hasattr(_var0, '_dynamo_dynamic_indices') == False:
+           return False
+       _var1 = L['b']
+       if not hasattr(_var1, '_dynamo_dynamic_indices') == False:
+           return False
+       if not ___is_grad_enabled():
+           return False
+       if ___are_deterministic_algorithms_enabled():
+           return False
+       if not ___is_torch_function_enabled():
+           return False
+       if not getattr(utils_device, 'CURRENT_DEVICE') == None:
+           return False
+       if not ___check_tensors(_var0, _var1, tensor_check_names=tensor_check_names
+           ):
+           return False
+       return True
+
+   compiled code:
+   def toy_example(a, b):
+       __temp_1 = __compiled_fn_0(a, b)
+       x = __temp_1[0]
+       if __temp_1[1]:
            return __resume_at_30_1(b, x)
-        return __resume_at_38_2(b, x)
+       return __resume_at_38_2(b, x)
 
-   def __resume_at_38_2(b, x):
-       return x * b
+Some names referenced in the code are:
+
+- Compiled functions, stored in the global namespace of the module containing the original function ``toy_example``. These include names like ``__compiled_fn_0`` / ``__resume_at_30_1`` / ``__resume_at_38_2``.
+
+- Clousure variables used for checking guards. The names can be accessed from ``guard.__code__.co_freevars``, and the values are stored in ``guard.__closure__``. These include names like ``___guarded_code`` / ``___is_grad_enabled`` / ``___are_deterministic_algorithms_enabled`` / ``___is_torch_function_enabled`` / ``utils_device`` / ``___check_tensors`` / ``tensor_check_names``.
+
+- Argument ``L`` of the ``guard`` function. This is a dict mapping the name of arguments of ``toy_example`` to its values. This is only available when the function is called, where the frame evaluation API comes into play. In short, ``L`` is a ``dict`` with structure of ``{'a': value_a, 'b': value_b}``. Therefore, you can see the code uses ``L['a']`` to refer to the input variable ``a``.
+
+The graph break is shown in the code of compiled ``toy_example``, where we have to use python interpreter to select the following graph to execute.
+
+Note that we pass a simple ``my_compiler`` function as the backend compiler, therefore the subgraph code ``__resume_at_38_2``, ``__resume_at_30_1``, and ``__compiled_fn_0`` remain python code. This can also be inspected (please ignore the function name, and only use the function signature and function body code):
```

Comment:
```suggestion
Note that we pass a simple ``my_compiler`` function as the backend compiler, therefore the subgraph code ``__resume_at_38_2``, ``__resume_at_30_1``, and ``__compiled_fn_0`` remain Python code. This can also be inspected (please ignore the function name, and only use the function signature and function body code):
```